src/Streamly/Internal/Data/Stream/StreamD/Exception.hs

```diff
@@ -262,11 +264,16 @@ bracket_ bef aft bet =
 -- | See 'Streamly.Internal.Data.Stream.IsStream.bracket'.
 --
 {-# INLINE_NORMAL bracket #-}
-bracket :: (MonadAsync m, MonadCatch m)
-    => m b -> (b -> m c) -> (b -> Stream m a) -> Stream m a
-bracket bef aft bet =
-    gbracket bef (inline MC.try) aft
-        (\a (e :: SomeException) _ -> aft a >> return (nilM (MC.throwM e))) bet
+bracket :: (MonadAsync m, MonadCatch m) =>
+       m b
+    -> (b -> m c)
+    -> (b -> m d)
+    -> (b -> m e)
+    -> (b -> Stream m a)
+    -> Stream m a
+bracket bef aft exc gc bet =
```

Comment:
Can we keep the order of handlers same as in `gbracket`?
`bef exe aft gt`?
Basically keep a consistent order?

Reply:
gbracket is the one that should be changed, I did not change it to keep the change minimal.